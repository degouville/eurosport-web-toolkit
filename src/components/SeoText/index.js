import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import { coreLightMinus1 } from 'src/colors';
import { H3 } from 'src/typography';

const StyledTitle = styled(H3)`
  color: ${coreLightMinus1};
`;
const StyledContent = styled.p`
  margin-top: 10px;
  font-size: 14px;
  color: ${coreLightMinus1};
`;

const SeoText = ({ title, subtitle, content }) => (
  <>
    <StyledTitle>{title}</StyledTitle>
    <StyledTitle>{subtitle}</StyledTitle>
    <StyledContent>{content}</StyledContent>
  </>
);

SeoText.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
};

export default SeoText;
