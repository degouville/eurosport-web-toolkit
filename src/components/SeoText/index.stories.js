import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import { SeoText } from '../..';

const seoTextStories = storiesOf('Components|SeoText', module);

seoTextStories.add('Seo text', () => (
  <SeoText
    title={text('title', 'SeoText title')}
    subtitle={text('subtitle', 'SeoText subtitle')}
    content={text(
      'content',
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus facilisis nisi sed massa laoreet, et aliquam metus faucibus. Integer sagittis lacinia dapibus. Praesent eros ante, rhoncus id lorem non, dignissim interdum nisl. In hac habitasse platea dictumst. Vivamus id efficitur erat, sed efficitur dui. Sed nisl tellus, cursus nec massa a, vulputate porta tortor. Mauris ullamcorper eros id dolor pharetra placerat. Nunc congue in nulla ut porttitor.'
    )}
  />
));
