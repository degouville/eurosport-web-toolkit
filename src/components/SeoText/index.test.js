import { shallow } from 'enzyme/build';
import React from 'react';

import SeoText from '.';

describe('SeoText', () => {
  it('renders SeoText', () => {
    expect(shallow(<SeoText title="title" subtitle="subtitle" content="content" />)).toMatchSnapshot();
  });
});
