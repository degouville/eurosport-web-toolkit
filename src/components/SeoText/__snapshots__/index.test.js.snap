// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeoText renders SeoText 1`] = `
<Fragment>
  <Styled(h3)>
    title
  </Styled(h3)>
  <Styled(h3)>
    subtitle
  </Styled(h3)>
  <Styled(p)>
    content
  </Styled(p)>
</Fragment>
`;
